src/test/java/api/item/ItemLastCheckInTests.java

```diff
@@ -0,0 +1,125 @@
+
+
+import java.net.MalformedURLException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import api.support.APITestContext;
+import api.support.APITests;
+import io.vertx.core.json.JsonObject;
+
+public class ItemLastCheckInTests extends APITests {
+
+  @Test
+  public void checkedInItemWithLoanShouldHaveLastCheckedInFields() throws InterruptedException,
+    MalformedURLException, TimeoutException, ExecutionException {
+
+    IndividualResource item = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource user = usersFixture.jessica();
+    UUID servicePointId = servicePointsFixture.cd1().getId();
+    DateTime checkInDate = DateTime.now();
+
+    loansFixture.checkOutByBarcode(item, user, new DateTime(DateTimeZone.UTC));
+
+    loansFixture.checkInByBarcode(item, checkInDate, servicePointId);
+    JsonObject actualItem = itemsClient.get(item.getId()).getJson();
+
+    JsonObject lastCheckIn = actualItem.getJsonObject("lastCheckIn");
+
+    assertThat(lastCheckIn.getString("dateTime"), is(checkInDate.toString()));
+    assertThat(lastCheckIn.getString("servicePointId"), is(servicePointId.toString()));
+    assertThat(lastCheckIn.getString("staffMemberId"), is(APITestContext.getUserId()));
+  }
+
+
+  @Test
+  public void shouldNotFailCheckInWhenInvalidOrEmptyLoggedInUserId() throws InterruptedException,
+    MalformedURLException, TimeoutException, ExecutionException {
+
+    IndividualResource item = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource user = usersFixture.jessica();
+    UUID servicePointId = servicePointsFixture.cd1().getId();
+    DateTime checkInDate = DateTime.now();
+    String defaultUserId = APITestContext.getUserId();
+
+    APITestContext.setUserId("null");
+    loansFixture.checkOutByBarcode(item, user, new DateTime(DateTimeZone.UTC));
+
+    loansFixture.checkInByBarcode(item, checkInDate, servicePointId);
+    JsonObject lastCheckIn = itemsClient.get(item.getId()).getJson().getJsonObject("lastCheckIn");
+
+    assertThat(lastCheckIn.getString("dateTime"), is(checkInDate.toString()));
+    assertThat(lastCheckIn.getString("servicePointId"), is(servicePointId.toString()));
+    assertThat(lastCheckIn.getString("staffMemberId"), is(nullValue()));
+
+    APITestContext.setUserId("");
+    loansFixture.checkInByBarcode(item, checkInDate, servicePointId);
+    lastCheckIn = itemsClient.get(item.getId()).getJson().getJsonObject("lastCheckIn");
+
+    assertThat(lastCheckIn.getString("dateTime"), is(checkInDate.toString()));
+    assertThat(lastCheckIn.getString("servicePointId"), is(servicePointId.toString()));
+    assertThat(lastCheckIn.getString("staffMemberId"), is(nullValue()));
+
+    APITestContext.setUserId(defaultUserId);
```

Comment:
What happens if the test fails, will this statement still be executed?

Reply:
Done.